src/lib.rs

```diff
@@ -292,6 +293,30 @@ impl From<JsonFrontendError> for Error {
 ///
 /// [`BpfProgram`]: type.BpfProgram.html
 pub fn apply_filter(bpf_filter: BpfProgramRef) -> Result<()> {
+    apply_filter_with_flags(bpf_filter, SeccompFlagset::new())
+}
+
+/// Apply a BPF filter to the all threads in the process.
+///
+/// # Arguments
+///
+/// * `bpf_filter` - A reference to the [`BpfProgram`] to be installed.
+///
+/// [`BpfProgram`]: type.BpfProgram.html
+pub fn apply_filter_all_threads(bpf_filter: BpfProgramRef) -> Result<()> {
```

Comment:
Having a close look at the [man page](https://man7.org/linux/man-pages/man2/seccomp.2.html), I see that the return value of the syscall when the tsync flag is set actually is different, and contains `the ID of the thread that caused the synchronization failure`.

We should also return a new type of error wrapping this thread id.

And while we're at it, change the `Error::Prctl` variant was used with both errors from the `NO_NEW_PRIVS` call and the `SET_FILTER`. Add a new variant to the error for filter installation errors.